packages/terser/src/index.js

```diff
@@ -48,6 +48,35 @@ log_verbose(`Running terser/index.js
   output: ${output}
   residual: ${residual}`);
 
+/**
+ * @param {string[]} residuleArgs
+ * @param {string} output
+ */
+function processSourceMapArg(residuleArgs, output) {
```

Comment:
this function seems like it would also benefit from some unit testing?

Reply:
it's a little hard to do, should, we'd have to export it from this file to be able to unit test, do you think exporting is a good idea?

Reply:
sure, exporting the function is fine I think, it's not like we need to document it as public API

Reply:
Aight, added some tests, required a bit of refactoring so needs a rereview